pkg/querier/tripperware/roundtrip.go

```diff
@@ -122,6 +123,12 @@ func NewQueryTripperware(
 			return RoundTripFunc(func(r *http.Request) (*http.Response, error) {
 				isQuery := strings.HasSuffix(r.URL.Path, "/query")
 				isQueryRange := strings.HasSuffix(r.URL.Path, "/query_range")
+				isBuildInfo := strings.HasSuffix(r.URL.Path, "/status/buildinfo")
+				// Build info is a static value, so we return here without sending to querier.
+				if isBuildInfo {
+					return buildInfoRoundTripper.RoundTrip(r)
+				}
+
```

Comment:
Maybe for another PR but would be nice to improve this and have some kind of routing logic here instead of all this `ifs`.